x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CreateSnapshotStep.java

```diff
@@ -61,7 +62,16 @@ public void onResponse(Boolean complete) {
 
             @Override
             public void onFailure(Exception e) {
-                listener.onFailure(e);
+                if (e instanceof SnapshotNameAlreadyInUseException) {
+                    // we treat a snapshot that was already created before this step as an incomplete snapshot. This scenario is triggered
+                    // by a master restart or a failover which can result in a double invocation of this step.
+                    logger.warn(
+                        "snapshot is already in-progress or in-use, this index will attempt to clean it up and retry snapshot creation"
```

Comment:
Can you format in a bit more context on this log message? Maybe something like `snapshot [somesnapshotname] is already ... for index [someindexname], will attempt ...` (you could even use the new pattern matching for `instanceof` syntax)?

Reply:
Done.